### DISCUSSION TITLE
how to check out midi device is connect?

### DISCUSSION THREAD

**Discussion by fareign** [tttapa/Control-Surface#924] [General]
for example :
I use the midi keyboard to control VMIX.
how to check out Vmix is connect or  not connect?

**Comment by tttapa** [comment]
As far as I know, this is not something the MIDI USB standard exposes to you:

> The USB-MIDI function exposes a single MIDIStreaming interface that is used by Host software to interact
and control the entire MIDI functionality of the function. Since all control messages are performed in-band
(buried into the MIDI data stream) there is no need for a separate control and status interface.

In-band, I suppose you could use the MIDI active sense message, but this is not of much use if you don't control the vMix software.  
You could try to deduce whether someone is listening to the messages you send by seeing whether the TX buffer drains, or check whether you get incoming MIDI messages, but this requires low-level access to the MIDI backend.

**Reply by fareign** [reply]
> As far as I know, this is not something the MIDI USB standard exposes to you:
> 
> > The USB-MIDI function exposes a single MIDIStreaming interface that is used by Host software to interact
> > and control the entire MIDI functionality of the function. Since all control messages are performed in-band
> > (buried into the MIDI data stream) there is no need for a separate control and status interface.
> 
> In-band, I suppose you could use the MIDI active sense message, but this is not of much use if you don't control the vMix software. You could try to deduce whether someone is listening to the messages you send by seeing whether the TX buffer drains, or check whether you get incoming MIDI messages, but this requires low-level access to the MIDI backend.

some soft doesn't have midi reply message.so I need to detect it. 
use which function?
